python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+            if np.max(lens['zl'] - lens_gal_df['redshift']) > 0.005:
+                if z_error is False:
+                    errors_string = errors_string + "\nLens Redshifts. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    z_lens_error = True
+
+            if np.max((lens_gal_df['majorAxis'] - lens['r_eff']/np.sqrt(1-lens['e']))) > 0.005:
+                if lens_major_axis_error is False:
+                    errors_string = errors_string + "\nLens Major Axis. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    lens_major_axis_error = True
+
+            if np.max((lens_gal_df['minorAxis'] - lens['r_eff']/np.sqrt(1+lens['e']))) > 0.005:
+                if lens_minor_axis_error is False:
+                    errors_string = errors_string + "\nLens Minor Axis. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    lens_minor_axis_error = True
+
+            if (lens_gal_df['sedFilepath'] != lens['lens_sed'].values[0]):
+                if lens_sed_error is False:
+                    errors_string = errors_string + "\nSED Filename. First error found in lens_gal_id: %i " % u_id
+                    errors_present = True
+                    lens_sed_error = True
+        
+        print('------------')
+        print('SNe Lenses direct catalog input Results:')
+
+        if errors_present is False:
+            print('Pass: All instance catalog values within 0.005 of lensed system inputs. ' +
+                  'All SED Filenames match.')
+        else:
+            raise CatalogError('\nFail:\n%s' % errors_string)
+
+        print('------------')
+
+        return
+
+    def compare_agn_lens_mags(self, spr_agn_df, spr_agn_lens_df):
+
+        """
+        This test compares the lens magnitudes.
+        """
+        
+        db = om10.DB(catalog=os.path.join(os.environ['TWINKLES_DIR'], 'data', 
+                                          'twinkles_lenses_v2.fits'), vb=False)
+
+        lens_mag_error = []
+
+        galSpecDir = 'galaxySED'
+        galDir = str(getPackageDir('sims_sed_library') + '/' + galSpecDir)
+        bandpassDict = BandpassDict.loadTotalBandpassesFromFiles(bandpassNames=['i'])
+        norm_bp = Bandpass()
+        norm_bp.imsimBandpass()
+
+        for lens_gal_row in spr_agn_lens_df.iterrows():
+
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_agn_df.query('lens_galaxy_uID == %i' % u_id)
+            use_lens = db.lenses['LENSID'][np.where(db.lenses['twinklesId'] == 
+                                                    spr_sys_df['twinkles_system'].iloc[0])[0]]
+            lens = db.get_lens(use_lens)
+            num_img = lens['NIMG']
+
+            test_sed = Sed()
+            test_sed.readSED_flambda('%s/%s' % (galDir, lens['lens_sed'][0]))
```

Comment:
Where is `lens['lens_sed']` coming from?  If it is coming from the InstanceCatalog, you should not need `galSpecDir` here (the InstanceCatalog should already refer to `galSpecDir`).  This may be the bug that I introduced with

https://github.com/LSSTDESC/sims_GCRCatSimInterface/commit/74a4f32b2d3e0b7c38b5a558dfd3d749b17adf26

(that commit takes the mapping with `sims.utils.defaultSpecMap` out of the getter method; I forgot that the sprinkler would not know about this change; I need to do some profiling to see if this really gets us a significant improvement in performance)

Reply:
`lens['lens_sed']` comes from the sprinkled lens catalog (OM10 lens plus added info like SEDs from CatSim) so this is needed here. Plus, I need to load the actual SED so I need the full filepath to the stored copy of the `sims_sed_library`.

Reply:
OK